src/15below.Deployment.Update/DictionaryExtensions.cs

```diff
@@ -0,0 +1,14 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace FifteenBelow.Deployment.Update
+{
+    public static class DictionaryExtensions
+    {
+        public static void AddOrDiscard(this Dictionary<string, object> dictionary, string key, string value, bool idSpecific = false)
+        {
+            if (!dictionary.Keys.Contains(key)) dictionary.Add(key, value);
+            if (dictionary.Keys.Contains(key) && idSpecific) dictionary[key] = value;
```

Comment:
Suggest you rename the idSpecific parameter to be "overwrite".
Also, can't you make this function Generic, rather than hardcoding to <string, object>